tests/integration/pilot/istioctl_test.go

```diff
@@ -172,3 +176,77 @@ func getPodID(i echo.Instance) (string, error) {
 
 	return "", fmt.Errorf("no workloads")
 }
+
+func TestProxyConfig(t *testing.T) {
+	framework.NewTest(t).
+		RequiresEnvironment(environment.Kube).
+		RunParallel(func(ctx framework.TestContext) {
+			ns := namespace.NewOrFail(ctx, ctx, namespace.Config{
+				Prefix: "istioctl-pc",
+				Inject: true,
+			})
+
+			var a echo.Instance
+			echoboot.NewBuilderOrFail(ctx, ctx).
+				With(&a, echoConfig(ns, "a")).
+				BuildOrFail(ctx)
+
+			istioCtl := istioctl.NewOrFail(t, ctx, istioctl.Config{})
+
+			podID, err := getPodID(a)
+			if err != nil {
+				ctx.Fatalf("Could not get Pod ID: %v", err)
+			}
+
+			var output string
+			var args []string
+			g := gomega.NewGomegaWithT(t)
+
+			args = []string{fmt.Sprintf("--namespace=%s", ns.Name()),
+				"pc", "bootstrap", podID}
+			output = istioCtl.InvokeOrFail(t, args)
+			jsonOutput := jsonUnmarshallOrFail(t, strings.Join(args, " "), output)
+			g.Expect(jsonOutput).To(gomega.HaveKey("bootstrap"))
+
+			args = []string{fmt.Sprintf("--namespace=%s", ns.Name()),
+				"pc", "cluster", podID, "-o", "json"}
+			output = istioCtl.InvokeOrFail(t, args)
+			jsonOutput = jsonUnmarshallOrFail(t, strings.Join(args, " "), output)
+			g.Expect(jsonOutput).To(gomega.Not(gomega.BeEmpty()))
+
+			args = []string{fmt.Sprintf("--namespace=%s", ns.Name()),
+				"pc", "endpoint", podID, "-o", "json"}
+			output = istioCtl.InvokeOrFail(t, args)
+			jsonOutput = jsonUnmarshallOrFail(t, strings.Join(args, " "), output)
+			g.Expect(jsonOutput).To(gomega.Not(gomega.BeEmpty()))
+
+			args = []string{fmt.Sprintf("--namespace=%s", ns.Name()),
+				"pc", "listener", podID, "-o", "json"}
+			output = istioCtl.InvokeOrFail(t, args)
+			jsonOutput = jsonUnmarshallOrFail(t, strings.Join(args, " "), output)
+			g.Expect(jsonOutput).To(gomega.Not(gomega.BeEmpty()))
+
+			args = []string{fmt.Sprintf("--namespace=%s", ns.Name()),
+				"pc", "route", podID, "-o", "json"}
+			output = istioCtl.InvokeOrFail(t, args)
+			jsonOutput = jsonUnmarshallOrFail(t, strings.Join(args, " "), output)
+			g.Expect(jsonOutput).To(gomega.Not(gomega.BeEmpty()))
+
+			args = []string{fmt.Sprintf("--namespace=%s", ns.Name()),
+				"pc", "secret", podID, "-o", "json"}
```

Comment:
We should also test the standard outputs to ensure that we catch any breaking changes in the config dump API.

Reply:
When you say "standard outputs" do you mean the tables?

I have struggled with this because the `pc <type> -o short` output is highly dependent on the test environment.

Reply:
Yeah, I would just verify the line count of the outputs are > 1.